packages/x-tailwindcss/demo/options.ts

```diff
@@ -7,7 +7,8 @@ import {
 } from '../src/types';
 
 export const options: PluginOptions = {
-  components({ theme }): ComponentsDefinition {
+  components(helpers): ComponentsDefinition {
+    const { theme } = helpers;
     return {
       '.icon': {
         display: 'flex',
```

Comment:
Here you can test to override css like this:
```js
      '.button': {
        '&.outlined': {
          '&.primary': {}
        }
      }
```

@javieri-empathy if you can take a look into the type, the `&.primary` is not inferred, like it does with the `&.outlined`, despite of the `mapColors` function has the right type. I think is because the `DeepPartial` in the `ComponentsDefinition` type, but I'm not sure

Reply:
And just by doing 2 small apparently harmless changes, the autocomplete suggestions work properly:

Use a bracketed access to define `ThemeColors`:

```ts
export type ThemeColors = typeof pluginTheme['colors'];
```

Use `Object.assign` to define `buttonOutlined`:

```ts
export function buttonOutlined(helpers: TailwindHelpers) {
  const { theme } = helpers;
  return {
    '&.outlined': Object.assign(
      {
        borderColor: theme('colors.neutral.50'),
        backgroundColor: theme('colors.neutral.0'),
        color: theme('colors.neutral.50'),

        '&:hover': {
          borderColor: theme('colors.neutral.50'),
          backgroundColor: theme('colors.neutral.50'),
          color: theme('colors.neutral.0')
        },

        '&:active': {
          borderColor: theme('colors.neutral.50'),
          backgroundColor: theme('colors.neutral.50'),
          color: theme('colors.neutral.0')
        },

        '&:disabled': {
          borderColor: theme('colors.neutral.25'),
          backgroundColor: theme('colors.neutral.10'),
          color: theme('colors.neutral.25')
        }
      },
      mapColors(
        color => ({
          borderColor: color['50'],
          backgroundColor: theme('colors.neutral.0'),
          color: color['50'],

          '&:hover': {
            backgroundColor: color['50'],
            borderColor: color['50'],
            color: theme('colors.neutral.0')
          },

          '&:active': {
            borderColor: color['50'],
            backgroundColor: color['50'],
            color: theme('colors.neutral.0')
          },

          '&:disabled': {
            borderColor: theme('colors.neutral.25'),
            backgroundColor: theme('colors.neutral.10'),
            color: theme('colors.neutral.25')
          }
        }),
        helpers
      )
    )
  };
}
```

I couldn't find any difference between the types using these new options, so my guess is that either Webstorm is doing something weird under the hood, or TS has a separate process for suggestions that does not follow the types one
![image](https://user-images.githubusercontent.com/68222542/192452821-56d78fc2-ffbb-46ef-82ac-5dd46f6893dc.png)
![image](https://user-images.githubusercontent.com/68222542/192452887-b6fd2c8a-cc0a-4767-8cc7-ccb8bad33362.png)